core/planner/planner.go

```diff
@@ -55,9 +55,14 @@ func (t *Planner) plan(rates api.Rates, requiredDuration time.Duration, targetTi
 		slotDuration := slot.End.Sub(slot.Start)
 		requiredDuration -= slotDuration
 
-		// slot covers more than we need, so lets start late
+		// slot covers more than we need, so shorten it
 		if requiredDuration < 0 {
-			slot.Start = slot.Start.Add(-requiredDuration)
+			// the first (if not single) slot should start as late as possible
+			if IsFirst(slot, plan) && len(plan) > 0 {
```

Comment:
Müsste es nicht plan > 1 heissen, oder ist der Planner wirklich leer, wenn nur noch ein Slot gerade aktiv ist?

Reply:
Hier wird der plan ja erst gebaut und das ist der allerletzte Slot *bevor* er hinzu gefügt wird. Wenns nur den gibt sit die aktuelle Länge 0.

Reply:
Ok. Dann passt das so. Ich hab mir die ganze Funktion jetzt noch mal durchgelesen. Dann gefällt mir "IsFirst()" nur so halb. Da du eigentlich in der "Plan()" Funktion auf der Liste "plan" rumarbeitest, diese könnte man über einen Mutex schützen (muss man hier nicht aber rein architektonisch), und die Liste die nicht fertig ist, geht an einen Helper raus. Daher meine Verwirrung. THeoretisch könnte sich ist First ja auch noch verändern wenn der plan noch nicht steht.

Macht es daher nicht sinn, das ganze beschneiden zu machen, wenn alle Slots im Plan sind? Also hinter der forschleife?

Reply:
Wir sind doch hinter der for schleife- der teuerste ist der letzte Slot. Mehr "hinter" kommt nicht. Und der `plan` ist eine lokale Variable:

	var plan api.Rates

Ich glaub du interpretierst da was rein?

Reply:
Gucke ich später noch mal. Es wird so funktionieren und ist mit den Stunden Slots ein riesen Schritt nach vorne. Dann mach es mal so. und wenn noch was auffallen sollte machen wir halt noch eine Iteration.